### PR TITLE
feat(#1264): add goa-calendar and goa-datepicker components

### PR DESCRIPTION
**Pull request by chrisolsen**

### CODE REVIEW

**Comment by Radha-Alberta** [comment]
- [ ] **Issue-1:** add examples in React App and Angular App
- [ ] **Issue-2:** Date focus is not working
- [ ] **Issue-3:** selecting unavailable date, month should change ? if yes, then selected date should move to available date
- [ ] **Issue-4:** month dropdown scroll not working
- [ ] **Issue-5:** opening dropdown, selected month should appear in dropdown list
- [ ] **Issue-6:** Home Key, End Key, PageUp, PageDown, Shift+pageUp, Shift+PageDown Key are not working as expected
- [ ] **Issue-7:** Mobile [ IOS-safari ]: improper selected date padding
- [ ] **Issue-8:** IOS-Safari : blue border absent on Today's date
- [ ] **Issue-9:**  see the steps to reproduce: [ https://www.loom.com/share/d5c81806cc13422fbb25b8d5de608653?sid=b614ee5a-7265-4610-a830-ae2894703383 ]

**Steps to Reproduce [ Issue-9 ]:**
1. Open Date picker with date attribute set as today's date.
2. Move selected date with arrow keys
3. move selected date to right (towards 31st August)
4. move that to 1st September [ NOTE: user won't be able to see selected date ]
5. click on any date from September [NOTE: month will change to October]

**Issue-2:**
image

**Issue-7:**
image


**Issue-8:**
image

**Comment by Radha-Alberta** [comment]
- [ ] Move to July by selecting unavailable date from August month. [No unavailable dates after 30 July]

**Steps to Reproduce:**
1. open Today's Date 
2. Click on 31st July [This will be unavailable date]
3. 31st July date will be selected. [ Issue: No Unavailable dates after 30 July. ]

- [ ] Above issue also exist for 31st December 2023 and March 2024 and June 2024

**Comment by Radha-Alberta** [comment]
**Issue-1:** PageUp not working properly
- Steps to Reproduce:
1. Go to September 4th, 2023
2. Press PageUp

**Current Output:** 5th August 2023
**Expected Output:** 4th August 2023
**Similar Dates with Issue:**  5th July 2023

**Issue-2:** Shift + page Up Not working as expected
- **Steps to Reproduce:**
1. Go to September 2nd, 2021
2. Press Shift + PageUp

**Current Output:** 1st September 2020
**Expected Output:** 2nd September 2020
**Similar Dates with Issue:** 3rd September 2020


**Issue-3:** Angular: [ https://www.loom.com/share/6569fcacdca04382b6d0ee4fc2615fcb?sid=228bfd0d-c201-4427-9b83-da7731c526f8 ]
1. Open Date picker
2. press Tab to shift focus to month dropdown
3. press shift+Tab to shift focus to previous element [ focus will move to last day of month, it should move to selected date]

**Issue-4:** Selecting date from date picker, input focus state should be active [just like date and time]( @Tom can you confirm this is a bug or not ?)

**Comment by Radha-Alberta** [comment]
- [ ] **Issue-1:** space key : opens the date picker but scrolls the DOM page as well
- [ ] **Issue-2:** In Min/Max : year dropdown has only current year
- [ ] **Issue-3:** month/year dropdown should close, if we close the date picker
- [ ] **Issue-4:** dropdown should open/close by pressing enter
- [ ] **Issue-5:** Shift+PageDown goes till 2027, it should go till 2028 (up to 5 years forward)
- [ ] **Issue-6:** there should be white underline border on the today's date,selected 
- [ ] **Issue-7:** extra double quotation below min/max date picker input


**Issue-6**
image
image

**Issue-7:**
image

**Comment by Radha-Alberta** [comment]
- [ ] **Issue-1:** Accessibility: selecting the date, it should announce date including month and year
- [ ] **Issue-2:** Accessibility: unable to select date with arrow keys
- [ ] **Issue-3:** not able to update date when NVDA is active.
- [ ] **Issue-4:** once date is updated with keyboard, month and year value gets cleared

https://www.loom.com/share/2aa2da6a7e3442c7863116ba7f1c0448?sid=8d49d5c4-9c61-445f-9963-24d6bec8bd76

**Steps to Reproduce:** [ NOTE: use only keyboard ]
1. Open date picker.
2. select any date from date picker.
3. reopen the date picker again [ **Issue-4:** month & year dropdown is cleared ]

File: libs/docs/src/components/common/Calendar.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      code={`
+		<goa-calendar></goa-calendar>
+    `}
+    />
+  </Tab>
+  <Tab label="React">
+    <CodeSnippet
+      lang="tsx"
+      code={`
+		<GoACalendar onChange() />
```
Comment (vanessatran-ddi) [inline libs/docs/src/components/common/Calendar.stories.mdx:97]: This example React code won't work. 
Suggestion change: <GoACalendar onChange={() => {}} />

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 2]
```diff
@@ -116,22 +114,21 @@
...
 
-    options.forEach((option: Option) => {
-      const label = option.label || option.value || "";
-      if (!width && maxCount < label.length) {
-        maxCount = label.length;
-        width = `${Math.max(20, maxCount + 12)}ch`;
-      }
-    });
-    return width;
+    // set width to longest item
```
Comment (thytran142) [inline libs/web-components/src/components/dropdown/Dropdown.svelte:122]: I checked on my local and looks like the width is different compared to alpha now. The dropdown's width seems to be smaller than the alpha's dropdown's width, even with a place holder text. 

image

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 3]
```diff
@@ -116,22 +114,21 @@
...
+    const width = options
```
Comment (vanessatran-ddi) [inline libs/web-components/src/components/dropdown/Dropdown.svelte:123]: I checked the alpha and this branch dropdown's width, the branch's dropdown's width is smaller than the alpha's version. 
image

File: libs/docs/src/components/common/DatePicker.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    />
+  </Tab>
+  <Tab label="React">
+    <CodeSnippet
+      lang="tsx"
+      code={`
+      <GoADatePicker 
+        onChange={(name: string, value: Date) => {}} 
+        value={new Date()} 
+        min={addDays(new Date(), -30)}
```
Comment (vanessatran-ddi) [inline libs/docs/src/components/common/DatePicker.stories.mdx:191]: Suggestion: I think we should add import
import {addDays} from "date-fns";

It's a bit unclear where addDays function is coming from, like our other storybook Inputs/Date, we have import from "date-fns".

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**